Mathlib/Analysis/Meromorphic/Divisor.lean

```diff
@@ -135,15 +134,15 @@ See `MeromorphicOn.exists_order_ne_top_iff_forall` and
 `h₂f₁` and `h₂f₂`.
 -/
 theorem divisor_smul {f₁ : 𝕜 → 𝕜} {f₂ : 𝕜 → E} (h₁f₁ : MeromorphicOn f₁ U)
-    (h₁f₂ : MeromorphicOn f₂ U) (h₂f₁ : ∀ z, (hz : z ∈ U) → (h₁f₁ z hz).order ≠ ⊤)
-    (h₂f₂ : ∀ z, (hz : z ∈ U) → (h₁f₂ z hz).order ≠ ⊤) :
+    (h₁f₂ : MeromorphicOn f₂ U) (h₂f₁ : ∀ z ∈ U, meromorphicOrderAt f₁ z ≠ ⊤)
+    (h₂f₂ : ∀ z ∈ U, meromorphicOrderAt f₂ z ≠ ⊤) :
     divisor (f₁ • f₂) U = divisor f₁ U + divisor f₂ U := by
   ext z
   by_cases hz : z ∈ U
-  · lift (h₁f₁ z hz).order to ℤ using (h₂f₁ z hz) with a₁ ha₁
-    lift (h₁f₂ z hz).order to ℤ using (h₂f₂ z hz) with a₂ ha₂
-    simp [h₁f₁, h₁f₂, h₁f₁.smul h₁f₂, hz, (h₁f₁ z hz).order_smul (h₁f₂ z hz), ← ha₁, ← ha₂,
-      ← WithTop.coe_add]
+  · lift meromorphicOrderAt f₁ z to ℤ using (h₂f₁ z hz) with a₁ ha₁
+    lift meromorphicOrderAt f₂ z to ℤ using (h₂f₂ z hz) with a₂ ha₂
```

Comment:
Can't you use `meromorphicOrderIntAt` here instead?

Reply:
I haven't introduced `meromorphicOrderIntAt` in the PR, as I've tried to make minimal changes. It may be a good idea, but for another PR then.